app/controllers/teams/users_controller.rb

```diff
@@ -1,36 +1,64 @@
 class Teams::UsersController < AuthenticatedController
-   before_action -> { authorize team, policy_class: TeamUsersPolicy }
-
-    def remove_confirm
-      if user_present_not_current_user(user)
-        render :remove_confirm      
-      else
-        render_401
-      end
-    end
-  
-    def destroy
-      if user_present_not_current_user(user)
-        team.users.delete(user)
-        flash[:success] = I18n.t('teams.users.remove.success', email: user.email)
-        redirect_to team_path(team.id)
-      else
-        render_401
-      end
-    end
-  
-    private
+  before_action -> { authorize team, policy_class: TeamUsersPolicy }
 
-    def user
-      @user ||= team.users.find_by(id: params[:id])
-    end
+  def new
+    @user = User.new
+  end
 
-    def user_present_not_current_user(user)
-      user.present? && user != current_user
+  def create
+    new_member.user_teams << UserTeam.create!(user_id: new_member.id, group_id: team.id)
+    flash[:success] = I18n.t('teams.users.create.success', email: member_email)
+    redirect_to team_path(team.id) and return
+  rescue ActiveRecord::RecordInvalid => e
+    flash[:error] = "Email '#{member_email}': " + e.record.errors.messages.values.flatten.pop
+    redirect_to new_team_user_path
+  end
+
+  def remove_confirm
+    if user_present_not_current_user(user)
+      render :remove_confirm
+    else
+      render_401
     end
+  end
 
-    def team
-      @team ||= Team.includes(:users).find(params[:team_id])
+  def destroy
+    if user_present_not_current_user(user)
+      team_and_users.users.delete(user)
+      flash[:success] = I18n.t('teams.users.remove.success', email: user.email)
+      redirect_to team_path(team.id)
+    else
+      render_401
     end
+  end
+
+  private
+
+  def member_email
+    user_params.require(:email).downcase
+  end
+
+  def new_member
+    @new_member ||= User.find_or_create_by!(email: member_email)
+  end
+
+  def user
+    @user ||= team.users.find_by(id: params[:id])
```

Comment:
**question:** would it make more sense to use `team_and_users` here? Or, in the interest of brevity, just calling it `team` and running `includes(:users)` all the time? I don't think the eager loading will ever have a real performance impact.

Reply:
changed it to team_and_users